jib-core/src/integration-test/java/com/google/cloud/tools/jib/api/ContainerizerIntegrationTest.java

```diff
@@ -218,7 +219,60 @@ public void testSteps_forBuildToDockerRegistry_multipleTags()
   }
 
   @Test
-  public void tesBuildToDockerRegistry_dockerHubBaseImage()
+  public void testSteps_forBuildToDockerRegistry_skipExistingDigest()
+      throws IOException, InterruptedException, ExecutionException, RegistryException,
+          CacheDirectoryCreationException {
+    System.setProperty(JibSystemProperties.SKIP_EXISTING_IMAGES, "true");
+
+    JibContainer image1 =
+        buildRegistryImage(
+            ImageReference.of("gcr.io", "distroless/java", DISTROLESS_DIGEST),
+            ImageReference.of("localhost:5000", "testimagerepo", "testtag"),
+            Collections.singletonList("testtag2"));
+
+    // Test that the initial image with the original tag has been pushed.
+    String imageReference = "localhost:5000/testimagerepo:testtag";
+    localRegistry.pull(imageReference);
+    assertDockerInspect(imageReference);
+    Assert.assertEquals(
+        "Hello, world. An argument.\n", new Command("docker", "run", "--rm", imageReference).run());
+
+    // Test that any additional tags have also been pushed with the original image.
+    String imageReference2 = "localhost:5000/testimagerepo:testtag2";
+    localRegistry.pull(imageReference2);
+    assertDockerInspect(imageReference2);
+    Assert.assertEquals(
+        "Hello, world. An argument.\n",
+        new Command("docker", "run", "--rm", imageReference2).run());
+
+    // Push the same image with a different tag, with SKIP_EXISTING_IMAGES enabled.
+    JibContainer image2 =
+        buildRegistryImage(
+            ImageReference.of("gcr.io", "distroless/java", DISTROLESS_DIGEST),
+            ImageReference.of("localhost:5000", "testimagerepo", "new_testtag"),
+            Collections.emptyList());
+
+    // Test that the pull request throws an exception, indicating that the new tag was not pushed.
+    String imageReference3 = "localhost:5000/testimagerepo:new_testtag";
+    try {
+      localRegistry.pull(imageReference3);
+      Assert.fail(
+          "jib.skipExistingImages was enabled and digest was already pushed, "
+              + "hence testtag2 shouldn't have been pushed.");
+    } catch (RuntimeException ignore) {
+      // As expected, registry throws exception that manifest is unknown.
+    }
+
+    // Test that both images have the same properties.
+    Assert.assertEquals(image1.getDigest(), image2.getDigest());
+    Assert.assertEquals(image1.getImageId(), image2.getImageId());
+
+    // Cleanup. Disable option to not disrupt other tests.
+    System.setProperty(JibSystemProperties.SKIP_EXISTING_IMAGES, "false");
```

Comment:
We use [`@Rule RestoreSystemProperties`](https://github.com/GoogleContainerTools/jib/blob/826b5bbff1192ec3bf411bf4d8a14bf52ad157c4/jib-core/src/test/java/com/google/cloud/tools/jib/configuration/BuildContextTest.java#L57) to clean up system properties, because cleaning this way is fragile unless it's wrapper around `try { ... } finally { ...}`.

Reply:
Just in case, I will "unresolve" conversations (include this) until I can verify actual changes. :)

Reply:
Ah, I was looking at an earlier commit. I will make undo "unresolving".